### PR TITLE
Refresh Kueue roadmap on the main page for 2025

### LINKED ISSUE #3377: The roadmap in README.md would benefit from an update
The roadmap in README.md currently refers to "main priorities for 2023"; we're rapidly approaching the end of 2024. For users it might be confusing to know whether or not the items on the priority list have already been completed or not.

### PR DESCRIPTION
**Pull request by mimowo**
#### What type of PR is this?

/kind documentation

#### Which issue(s) this PR fixes:

Fixes #3377

#### Special notes for your reviewer:

- This is just a roadmap, be aware the plans may change, we can update it later
- feel free to propose additional items to include

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

**Comment by mimowo** [comment]
@tenzen-y @kannon92 @mwielgus @mwysokin @dgrove-oss PTAL

**Comment by mimowo** [comment]
/hold 
To avoid merging prematurely, I would prefer to get a couple of LGTMs from you folks

File: README.md [hunk 1]
```diff
@@ -97,22 +97,29 @@ presents the detailed design of the controller.
...
+  * better accuracy of scheduling by tighter integration with kube-scheduler [#3755](https://github.com/kubernetes-sigs/kueue/issues/3755)
+  * reduce friction by defaulting the PodSet annotations [#3754](https://github.com/kubernetes-sigs/kueue/issues/3754)
+- Productization of the Kueue dashboard [#940](https://github.com/kubernetes-sigs/kueue/issues/940)
+- Support Hierarchical Cohorts with FairSharing [#3759](https://github.com/kubernetes-sigs/kueue/issues/3759)
+- Improved support for AI inferance, including:
+  * partial preemption of serving workloads [#3762](https://github.com/kubernetes-sigs/kueue/issues/3762)
+  * LeaderWorkerSet support [#3232](https://github.com/kubernetes-sigs/kueue/issues/3232)
+- Progress towards the stable API (v1beta2) [#768](https://github.com/kubernetes-sigs/kueue/issues/768)
+
+Long-term aspirational goals:
```
Comment (kannon92) [inline README.md:116]: I don't know if I would consider v1beta1 that much difference in stability than v1beta2. Can we add a long-term goal for v1 release of the API?

File: README.md [hunk 2]
```diff
@@ -97,22 +97,29 @@ presents the detailed design of the controller.
...
-- Integration with cluster-autoscaler for guaranteed resource provisioning
-- Integration with common custom workloads [#74](https://github.com/kubernetes-sigs/kueue/issues/74):
-  - Kubeflow (TFJob, MPIJob, etc.)
-  - Spark
-  - Ray
-  - Workflows (Tekton, Argo, etc.)
-
-These are features that we aim to have in the long-term, in no particular order:
-
+High-level overview of the main priorities for 2025:
```
Comment (kannon92) [inline README.md:100]: What do you think about adding DRA?

File: README.md (hunk 1 above)
Comment (mimowo) [reply README.md:116]: Sure, I pushed an update with the point - if you open such issue I can link it

File: README.md (hunk 2 above)
Comment (mimowo) [reply README.md:100]: Sure we can add, but maybe to the long-term goals, to be less commital?

Comment (mimowo) [reply README.md:100]: for now I added it to the bottom section

File: README.md (hunk 1 above)
Comment (kannon92) [reply README.md:116]: #3476

Comment (mimowo) [reply README.md:116]: awesome, done

**Review by tenzen-y — COMMENTED** [review]
Thank you for reflecting our recent discussions!
/lgtm
/approve

/hold cancel

**Pull request merged**